### PR TITLE
[MRG] Start glossary

### LINKED ISSUE #390: glossary
Adding a glossary with commonly added terms would be super useful. I can take an initial stab at this

### PR DESCRIPTION
**Pull request by jasmainak**
closes #390 

I started a glossary, please feel free to correct and/or add new items.

I just want to get the ball rolling and we can make this better in future PRs

cc @Moham-Amin @rythorpe

### CODE REVIEW

File: doc/glossary.rst [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        An artificial cell or population of artificial cells that only produces
+        spikes according to a given pattern and is used to initiate and sustain
+        cell spiking in the network
+
+    gid
+        The "global" cell ID. Global refers to the fact that the cell ID
+        is unique even if the simulation is split across cores.
+
+    distal
+        anatomical location that is further away from the soma
```
Comment (ntolley) [inline doc/glossary.rst:24]:
```suggestion
        Anatomical location that is further away from the soma.
        Distal drives refer to extrinsic connections to the distal dendrites of
        the cortical column, which often originate from direct thalamic nuclei.
```

File: doc/glossary.rst [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    proximal
+        anatomical location that is closer to the soma
```
Comment (ntolley) [inline doc/glossary.rst:27]:
```suggestion
        Anatomical location that is closer to the soma.
        Proximal drives refer to extrinsic connections to the proximal dendrites
        of the cortical column, which often originate from indirect thalamic
        nuclei, as well as other cortical regions.
```

**Comment by ntolley** [comment]
Leaving some words here for me to commit later, or for someone else if they want to start filling in definitions. Obviously we don't want to go so far as to define every basic neuroanatomy term (dendrite, soma, synapse, etc.), but certain things like cell type names and their primary role may not be familiar to neuroscientists that aren't obsessed with the cortex like us.
```
pyramidal: excitatory cell type, AMPA, NMDA, long apical dendrite
basket: inhibitory cell type, GABAa, GABAb
dipole
local field potential
somatic voltage/current
spikes
space constant
src_gid
target_gid
soma: really powerful location for inhibitory synaptic connections
gbar
dt

```

**Comment by jasmainak** [comment]
yeah, I would say if you can find something easily by googling or on wikipedia, perhaps not worth an entry. Also `src_gid` and `target_gid` should be addressed in docstrings of respective functions.

File: doc/glossary.rst [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+please consider creating a new issue or opening a pull request to add it.
+
+.. glossary::
+    :sorted:
+
+
+    drive
+        An artificial cell or population of artificial cells that only produces
+        spikes according to a given pattern and is used to initiate and sustain
+        cell spiking in the network
```
Comment (rythorpe) [inline doc/glossary.rst:17]: Is "drive" an artificial cell or the synaptic spikes that emerge from an artificial cell? This is nuanced, but I think of "drive" as modeling a physiological process rather than a physiological structure.

File: doc/glossary.rst [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+    gid
+        The "global" cell ID. Global refers to the fact that the cell ID
+        is unique even if the simulation is split across cores.
```
Comment (rythorpe) [inline doc/glossary.rst:21]:
```suggestion
        The "global" cell ID. A gid is assigned to each cell in the
        network as well as artificial drive cells. Global refers to
        the fact that the cell ID is unique even if the simulation
        is split across cores.
```

File: doc/glossary.rst [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+    distal
+        Anatomical location that is further away from the soma.
+        Distal drives refer to extrinsic connections to the distal dendrites of
+        the cortical column, which often originate from direct thalamic nuclei.
```
Comment (rythorpe) [inline doc/glossary.rst:26]:
```suggestion
        the cortical column, which often originate from indirect thalamic nuclei.
```

File: doc/glossary.rst [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+    proximal
+        Anatomical location that is closer to the soma.
+        Proximal drives refer to extrinsic connections to the proximal dendrites
+        of the cortical column, which often originate from indirect thalamic
+        nuclei, as well as other cortical regions.
```
Comment (rythorpe) [inline doc/glossary.rst:32]:
```suggestion
        of the cortical column, which often originate from direct thalamic
        nuclei.
```

**Comment by rythorpe** [comment]
Thanks for starting this @jasmainak! I made a few suggestions above and am happy to contribute in later PRs.

File: doc/glossary.rst (hunk 3 above)
Comment (jasmainak) [reply doc/glossary.rst:17]: agreed, and we may even want to avoid saying "artificial cells" ... it's bringing another new term to explain it. Can you make a suggested edit?

File: doc/glossary.rst (hunk 6 above)
Comment (jasmainak) [reply doc/glossary.rst]: is "direct"/"indirect" an anatomical location? I often get confused by this kind of phrasing that seems quite common on neuroscience ... you google "direct thalamic nuclei" and you get "direct pathway" and "thalamic nuclei" but no "direct thalamic nuclei" ...

File: doc/glossary.rst (hunk 3 above)
Comment (rythorpe) [reply doc/glossary.rst:17]:
```suggestion
    drive
        An exogenous source of excitatory action potentials (spikes)
        that synapse onto neuronal compartments, perturbing the network.
```

Comment (jasmainak) [reply doc/glossary.rst]: okay I deleted the second sentence to keep it simple for now. Thanks for the suggestion!

File: doc/glossary.rst (hunk 6 above)
Comment (rythorpe) [reply doc/glossary.rst]: > is "direct"/"indirect" an anatomical location? I often get confused by this kind of phrasing that seems quite common on neuroscience ... you google "direct thalamic nuclei" and you get "direct pathway" and "thalamic nuclei" but no "direct thalamic nuclei" ...

I'm not convinced that "direct" and "indirect" are the proper terms to use here. Maybe we should stick with "lemniscal" and "non-lemniscal" as described in the HNN-GUI tutorials.

This distinction does not separate two anatomical regions, rather, it's an informal classification scheme that some anatomists use to categorize thalamic nuclei based on their afferent projections into cortex. For instance, the ventral posteromedial nucleus (VPM) is part of the lemniscal pathway that has bulk projections to the granular layer of S1. Conversely, the posterior medial nucleus (POm) has dispersed projections (that are not part of the lemniscal pathway) to numerous cortical regions that happen to include the superficial layers of S1. This might be a good place for @stephanie-r-jones to weigh in.

**Review by rythorpe — APPROVED** [review]
Looks good to me, though what do you think of having Steph review and rubber-stamp the glossary terms in our next HNN meeting prior to our 0.2 release? I think she's going to have better intuition for what areas our user base is most likely to find confusing.

**Comment by jasmainak** [comment]
Sounds good. I'm thinking of having the glossary terms to be more accessible to a first year graduate student rather than be textbook definitions. More like "notes" / "added explanations". How much can we dumb it down vs be technically very precise?

**Comment by rythorpe** [comment]
> Sounds good. I'm thinking of having the glossary terms to be more accessible to a first year graduate student rather than be textbook definitions. More like "notes" / "added explanations". How much can we dumb it down vs be technically very precise?

I'm totally with you there. Anything here that can be said in <=1 sentence should be. I just want to be careful that we don't accidentally stumble into using jargon that comes with a lot of historical baggage.

**Comment by jasmainak** [comment]
@rythorpe what's our plan here? Do you want to discuss this in the next meeting or do we want to hold this off until later?

**Comment by jasmainak** [comment]
@rythorpe I rebased it now!

**Comment by rythorpe** [comment]
Looks good, thanks @jasmainak!
